From 6b4d8eecab8c8186a383798543d68c24af239ca5 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Thu, 8 Apr 2010 01:17:09 +0800
Subject: [PATCH 11/17] also mention missing objects

Junio's suggested text was not used, but the issue (no object found) was
addressed.

Based on:

  Message-ID: <7vskdss3ei.fsf@alter.siamese.dyndns.org>

  > +General Request Processing
  > +--------------------------
  > +
  > +Except where noted, all standard HTTP behavior SHOULD be assumed
  > +by both client and server.  This includes (but is not necessarily
  > +limited to):
  > +
  > +If there is no repository at $GIT_URL, the server MUST respond with
  > +the '404 Not Found' HTTP status code.

  We may also want to add

      If there is no object at $GIT_URL/some/path, the server MUST respond
      with the '404 Not Found' HTTP status code.

  to help dumb clients.
---
 Documentation/technical/http-protocol.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Documentation/technical/http-protocol.txt b/Documentation/technical/http-protocol.txt
index a13e03d..3858209 100644
--- a/Documentation/technical/http-protocol.txt
+++ b/Documentation/technical/http-protocol.txt
@@ -105,7 +105,9 @@ by both client and server.  This includes (but is not necessarily
 limited to):
 
 If there is no repository at $GIT_URL, the server MUST NOT respond with
-'200 OK' and a valid info/refs response.  A server SHOULD respond with
+'200 OK' and a valid info/refs response.  Also, if the resource pointed
+to by a location matching $GIT_URL does not exist, the server MUST NOT
+respond with '200 OK'.  A server SHOULD respond with
 '404 Not Found', '410 Gone', or any other suitable HTTP status code
 which does not imply the resource exists as requested.
 
-- 
1.8.4.rc4.527.g303b16c

